### PR TITLE
Prefs [TASK][27928] Remote data source

### PR DESCRIPTION
**Pull request by NguyenTaTrung**
## Related Tickets
- [#Task_27928](https://edu-redmine.sun-asterisk.vn/issues/27928)

## WHAT
- Create remote data source

## HOW
- Create model
- Create bind ready to transfer data

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Reviewer1 (TrungNT) | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | Do the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li> 
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/sun/trainingcourse/anticovid_19/data/resource/CovidDataSource.kt [hunk 1]
```diff
@@ -6,6 +6,6 @@ interface CovidDataSource {
     }
 
     interface Remote {
-
+        fun getCovidInformation()
```
Comment (aug16vcc) [inline app/src/main/java/sun/trainingcourse/anticovid_19/data/resource/CovidDataSource.kt:9]: Không trả về cái gì à?

File: app/src/main/java/sun/trainingcourse/anticovid_19/data/resource/CovidRepository.kt [hunk 2]
```diff
@@ -1,5 +1,15 @@
 package sun.trainingcourse.anticovid_19.data.resource
 
-class CovidRepository : CovidDataSource.Local, CovidDataSource.Remote {
+class CovidRepository private constructor(
+    private val remote: CovidDataSource.Remote
+): CovidDataSource.Local, CovidDataSource.Remote {
 
+    override fun getCovidInformation() {
+    }
```
Comment (aug16vcc) [inline app/src/main/java/sun/trainingcourse/anticovid_19/data/resource/CovidRepository.kt:8]:
```suggestion
    override fun getCovidInformation(): SomeType = TODO()
```
vì `TODO()` trả về Nothing

File: app/src/main/java/sun/trainingcourse/anticovid_19/utils/Constants.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package sun.trainingcourse.anticovid_19.utils
+
+object ModelConst {
+    const val NEW_CONFIRMED = "NewConfirmed"
+    const val NEW_DEATHS = "NewDeaths"
+    const val NEW_RECOVERED = "NewRecovered"
+    const val TOTAL_CONFIRMED = "TotalConfirmed"
+    const val TOTAL_DEATHS = "TotalDeaths"
+    const val TOTAL_RECOVERED = "TotalRecovered"
+}
```
Comment (aug16vcc) [inline app/src/main/java/sun/trainingcourse/anticovid_19/utils/Constants.kt]: Thiếu dòng trống

File: app/src/main/java/sun/trainingcourse/anticovid_19/data/resource/CovidDataSource.kt (hunk 1 above)
Comment (NguyenTaTrung) [reply app/src/main/java/sun/trainingcourse/anticovid_19/data/resource/CovidDataSource.kt:9]: Có chú nhưng đến task sau mới thêm

**Review by tuanpa-0908 — APPROVED** [review]

**Pull request merged**